src/Microsoft.AspNetCore.Server.Kestrel/Infrastructure/Components/StreamFactory.cs

```diff
@@ -0,0 +1,14 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Server.Kestrel.Http;
+
+namespace Microsoft.AspNetCore.Server.Kestrel.Infrastructure
+{
+    public class StreamFactory : DefaultComponentFactory<Streams>
```

Comment:
Why are the default factories so complicated if the default pool size is zero? I don't think our default factories should have any collections at all.

I think it may be better to move the pooling factories into another project to show it can be easily done and to simplify the main code base.